src/psyclone/psyir/nodes/intrinsic_call.py

```diff
@@ -2720,7 +3398,16 @@ class Intrinsic(IAttr, Enum):
                 types=Reference,
                 arg_names=(("source",),)),
             optional_args={"kind": DataNode},
-            return_type=None,
+            return_type=lambda node: (
+                ArrayType(ScalarType(
```

Comment:
You may already have a helper that does this?

Reply:
I don't think so - this always returns a rank 1 array and I don't think any of the helpers can be persuaded to do that for this intrinsic.